src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/InfrastructureFormula.kt

```diff
@@ -59,16 +72,17 @@ class InfrastructureFormula<out T : VirtualUsers>(
         }
 
         val network = networkProvisioning.get()
-        val provisionJira = executor.submitWithLogContext("jira") {
-            overrideJiraNetwork(network).provision(
+
+        val provisionedJira =  userProvisionedJira ?: executor.submitWithLogContext("jira") {
```

Comment:
Please explain how is this simpler than https://github.com/atlassian/aws-infrastructure/pull/58

Reply:
It is one more block in API. Feel free to continue this way. I just need to have one way to  achieve that.

Reply:
I get it, it's choosing the `JiraFormula` impl vs choosing the `InfrastructureFormula` constructor. Neither is strictly simpler than the other.
*But* the API already forces the choice of `JiraFormula` impl, so it fits the current patterns. And also this impl requires internal null checking and additional branches of execution (hidden `if`s) and more internal states to track and debug.
Also the parameter-expanding design is a poor trajectory for API evolution, because there's more and more permutations of optional parameters. Imagine adding another optional feature, now we'd have 4 constructors, then 8.

Reply:
So I hope you're aware of the intricacies, alternatives and hidden costs. I'm gonna unblock the PR

Reply:
Given the choice, you merged this version, despite all of the pros and cons. Could you share the rationale?